scipy/optimize/_minimize.py

```diff
@@ -206,10 +210,12 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
 
         where ``intermediate_result`` is a keyword parameter containing an
         `OptimizeResult` with attributes ``x`` and ``fun``, the present values
-        of the parameter vector and objective function. Note that the name
-        of the parameter must be ``intermediate_result`` for the callback
-        to be passed an `OptimizeResult`. These methods will also terminate if
-        the callback raises ``StopIteration``.
+        of the parameter vector and objective function. If ``jac`` or ``hess`` 
+        are specified as callable they will also be present in 
+        ``intermediate_result`` as attributes ``jac`` and ``hess``.
```

Comment:
You had written:

> These estimates might be available, but I checked how the final results are usually returned and there for instance the hessian is only provided, if a function to evaluate the hessian analytically is given.

but the following methods report a `jac` estimate even if they are not provided a callable: `'cg', 'bfgs', 'l-bfgs-b', 'tnc', 'slsqp'`

<details>

```python3
from scipy import optimize
methods = ['cg', 'bfgs', 'l-bfgs-b', 'tnc', 'slsqp', 'trust-constr']
for method in methods:
    res = optimize.minimize(optimize.rosen, [1, 2, 3], method=method)
    print(f"{method}: {getattr(res, 'jac', None)}")
```
`jac` reported by ` trust-constr` is an empty list, which is a different issue...

</details>

`tnc` and `slsqp` don't have the `intermediate_result` callable interface, but I'd expect `'cg', 'bfgs', 'l-bfgs-b'` to have `jac` in the `intermediate_result` even if the user doesn't provide a callable.

But you're right, the other methods don't seem to report `hess` unless `hess` is a callable.

```suggestion
        of the parameter vector and objective function. If ``jac`` or ``hess`` 
        are available in the returned result object, they will also be present in
        ``intermediate_result`` as attributes ``jac`` and ``hess``.
```